app/routes/user/endorsements.server.ts

```diff
@@ -0,0 +1,151 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+
+import { tables } from '@architect/functions'
+import { getUser } from '../__auth/user.server'
+
+// models
+export type EndorsementRequest = {
+  endorserEmail: string
+  endorserSub: string
+  requestorEmail: string
+  requestorSub: string
+  created: number
+  status: EndorsementStates
+}
+
+export type EndorsementStates = 'approved' | 'pending' | 'rejected' | 'reported'
+
+export class EndorsementsServer {
+  #sub: string
+
+  private constructor(sub: string) {
+    this.#sub = sub
+  }
+
+  static async create(request: Request) {
+    const user = await getUser(request)
+    if (!user)
+      throw new Response('not signed in', {
+        statusText: 'not signed in',
+        status: 403,
+      })
+    return new this(user.sub)
+  }
+
+  // FIXME: Requests should probably only be created with endorsers as users who are in the submitters group
+  async createEndorsementRequest(request: EndorsementRequest) {
+    const db = await tables()
+    await db.circular_permission_requests.put(request)
+  }
```

Comment:
Add Cognito group membership check based on access token claims here.

Reply:
I added a group check, but its checking against other users. I dont use the claims though, let me know what you think